### ISSUE TITLE
[3.0.0-beta.1] White background appears when .hide() and .show() called again

### ISSUE THREAD

**Issue by natew** [electron/electron#13378]
* Electron Version: 3.0.0-beta.1
* Operating System (Platform and Version): OS X High Sierra
* Last known working Electron version: 2.0.3

**Expected Behavior**

Using a transparent window, then calling .hide() and .show(). It should stay transparent.

**Actual behavior**

In 3.0.0-beta.1, it seems to turn the background white. If I trigger the DOM to change afterwards, the white background goes away. I am attached to it using devtools debugging, I will verify if that affects it as I had seen that bug before.

**To Reproduce**

I'll attempt to setup a repo here soon.

You can fork [electron-quick-start](https://github.com/electron/electron-quick-start) and include a link to the branch with your changes.

**Comment by codebytere** [comment]
Can't reproduce with current `master` running MacOS High Sierra; here is my current code:

<details>

```
const { app, BrowserWindow, Menu } = require('electron')
const path = require('path')
const url = require('url')

let win

function createWindow () {
  win = new BrowserWindow({
    name: 'My app window',
    width: 800,
    height: 600,
    transparent: true,
    frame: false,
    toolbar: false
  })

  const template = [
    {
      label: 'help',
      submenu: [
        {
          label: 'CAUSE ERROR',
          click () { win.isVisible() ? win.hide() : win.show() }
        }
      ]
    }
  ]

  const menu = Menu.buildFromTemplate(template)
  Menu.setApplicationMenu(menu)

  // and load the index.html of the app.
  win.loadURL(url.format({
    pathname: path.join(__dirname, 'index.html'),
    protocol: 'file:',
    slashes: true
  }))

  win.on('closed', () => { win = null })
}

app.on('ready', createWindow)

app.on('window-all-closed', () => {
  if (process.platform !== 'darwin') {
    app.quit()
  }
})

app.on('activate', () => {
  if (win === null) {
    createWindow()
  }
})
```
</details>

**Comment by ckerr** [comment]
@natew, does this still occur for you in 3.0.0-beta.3? Or, if you don't want to get trapped in that "try it again now" cycle, could you fork https://github.com/electron/electron-quick-start and make a test case that exercises this bug?

**Comment by natew** [comment]
Looks fixed now! Awesome, 3.0 actually feels snappier!

**Comment by codebytere** [comment]
awesome, thanks! Going to go ahead and close it in that event :)
